integration_tests/test_list_06.py

```diff
@@ -0,0 +1,48 @@
+from ltypes import i32, f64
+
+def check_mat_and_vec(mat: list[list[f64]], vec: list[f64]):
+    rows: i32 = len(mat)
+    cols: i32 = len(vec)
+    i: i32
+    j: i32
+
+    for i in range(rows):
+        for j in range(cols):
+            assert mat[i][j] == float(i + j)
+
+    for i in range(cols):
+        assert vec[i] == 2 * float(i)
+
+def test_list_of_lists():
+    tensor: list[list[list[f64]]] = []
+    mat: list[list[f64]] = []
+    vec: list[f64] = []
+    rows: i32 = 10
+    cols: i32 = 5
+    i: i32
+    j: i32
+    k: i32
+
+    for i in range(rows):
+        for j in range(cols):
+            vec.append(float(i + j))
+        mat.append(vec)
+        vec.clear()
+
+    for i in range(cols):
+        vec.append(2 * float(i))
+
+    check_mat_and_vec(mat, vec)
+
+    for k in range(rows):
+        tensor.append(mat)
+        for i in range(rows):
+            for j in range(cols):
+                mat[i][j] += float(1)
+
+    for k in range(rows):
+        for i in range(rows):
+            for j in range(cols):
+                assert mat[i][j] - tensor[k][i][j] == rows - k
```

Comment:
@certik This check would fail with CPython because in CPython lists are shallow copied but in LPython we deepcopy lists (see https://github.com/lcompilers/lpython/pull/835#discussion_r932470488). So I haven't enabled `cpython` for now in the tests. We can make it work with CPython by calling `copy.deepcopy` and ignoring the same in LPython as we anyways do deepcopy by default. What do you say?

Reply:
Yes, using `deepcopy` should work.